### PR TITLE
Controller tests

### PR DESCRIPTION
**Pull request by ibugithub**
ReservationsController Test 🚀
Description: Comprehensive tests covering all Reservation actions. ✅

Reservation Model Test 🧪
Description: Add a new test for the Reservation model. 📊

In this branch I did the following:

Git Conflict Resolution 🔀
Description: resolved merge conflicts for a harmonious codebase. 🌐🤝

Secret Key Trouble 🔐❗
Description: Battled the "key not found" error and secured the secret key base. 🛡️🔑

Credentials.yml.enc Update 🔄
Description: Safely edited the credentials file and added a fresh secret key base. 📜🔒

Restoring Application Security 🛠️
Description: Successfully restored security by ensuring Rails has its secret key. 🚧🏰

### CODE REVIEW

**Review by MoyasiGinko — CHANGES_REQUESTED** [review]
### Hey @ibugithub 👋 
Great job so far, but there are still some issues to fix before you can move forward.



### Required Changes:
- [ ] Please fix all the linters error, as it will create an issue when merged into the development branch.
- [ ] [OPTIONAL] Please attach a screenshot of the running tests that you have implemented for the record along with the PR.

Thank you and good luck. ⭐

**Comment by ibugithub** [comment]
Moyasi, Thank you so much man, I can't share the screenshot now. After merging my branch to dev All of the test cases failed because of the change that happened on the user table. So I am now fixing all the new errors. Then I'll provide the screenshot.

File: config/master.key
Comment (lincoln1883) [inline config/master.key:1]: This should never happen.

File: config/environments/development.rb [hunk 1]
```diff
@@ -63,4 +63,6 @@
   # Uncomment if you wish to allow Action Cable access from any origin.
   # config.action_cable.disable_request_forgery_protection = true
   config.action_mailer.default_url_options = { host: 'localhost', port: 3001 }
+  config.secret_key_base = Rails.application.credentials.secret_key_base
```
Comment (lincoln1883) [inline config/environments/development.rb:66]: Why was this line necessary?
